### PR TITLE
ADD ordinals and unittests for new pcs

### PR DESCRIPTION
**Pull request by KEggensperger**
This PR adds the option to write ordinal parameters and adds some more unittests to test the latest pcs format.

### CODE REVIEW

File: test/io/test_pcs_converter.py [hunk 1]
```diff
@@ -339,7 +339,65 @@ def test_build_new_forbidden(self):
...
+        cs = ConfigurationSpace()
+        a = UniformFloatHyperparameter("a", 0, 1, 0.5)
+        b = UniformIntegerHyperparameter("b", 0, 10, 5)
+        cs.add_hyperparameter(a)
+        cs.add_hyperparameter(b)
+        cond = GreaterThanCondition(a, b, 5)
+        cs.add_condition(cond)
+
+        value = pcs_new.write(cs)
+        self.assertIn(expected, value)
```
Comment (mfeurer) [inline test/io/test_pcs_converter.py]: I'm wondering why you're not checking for equality here (and in all the other checks as well).

File: test/io/test_pcs_converter.py [hunk 2]
```diff
@@ -436,5 +494,5 @@ def test_read_write(self):
             fh.write(pcs_string)
         with open(name, 'r') as fh:
             pcs_new = pcs.read(fh)
-        assert (pcs_new, cs)
+        self.assertTrue(pcs_new, cs)
```
Comment (mfeurer) [inline test/io/test_pcs_converter.py]: I agree that the old check did nothing, but I think the new one is broken, too. I think you rather want `self.assertEqual(pcs_new, cs)` as the current one only check that the pcs_new string is not empty.

**Comment by mlindauer** [comment]
ping....

File: test/io/test_pcs_converter.py (hunk 1 above)
Comment (KEggensperger) [inline test/io/test_pcs_converter.py]: Agree

File: test/io/test_pcs_converter.py (hunk 2 above)
Comment (KEggensperger) [inline test/io/test_pcs_converter.py]: @mfeurer Do you have any further ideas:
Fixing that check made it fail. I do not know why.

Looking at the output of print ```pcs_new``` and ```cs``` I could not find a difference.

**Comment by mfeurer** [comment]
The failing test is related to #25.

**Pull request merged**